packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,19 +37,94 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+/// A position that the [FloatingActionButton] can dock to in a [Scaffold].
+/// 
+/// [getOffset] 
+class FabPosition {
+  /// End-aligned FAB, floating at the bottom of the screen.
+  /// 
+  /// This is the default alignment of FABs in Material apps.
+  static const FabPosition endFloat = const _EndFloatFab();
+
+  /// Centered FAB, floating at the bottom of the screen.
+  static const FabPosition centerFloat = const _CenterFloatFab();
+
+  const FabPosition(this.getOffset);
+
+  /// Places the FAB with the layout parameters of the [Scaffold].
+  /// 
+  /// [scaffoldSize] is the size of the whole [Scaffold].
+  /// [fabSize] is the size of the [FloatingActionButton] to be placed.
+  /// [snackBarSize] is the size of the Scaffold's [SnackBar] (if available).
+  /// [bottomSheetSize] is the size of the [BottomSheet] (if available).
+  /// [textDirection] is the direction used by the Scaffold (either RTL or LTR).
+  /// [contentBottom] is the height from the top where the Scaffold's body ends.
+  /// [horizontalPadding] is padding the Scaffold would like the FAB to observe.
+  final Offset Function(Size scaffoldSize, Size fabSize, Size snackBarSize, Size bottomSheetSize, TextDirection textDirection, double contentBottom, double horizontalPadding) getOffset;
```

Comment:
Let's use a typedef here so that people can indirect this without having to duplicate the entire signature

Reply:
Based on discussion offline, changed this to using inheritance and taking a ScaffoldPrelayoutGeometry object as a parameter.